kai/reactive_codeplanner/task_manager/task_manager.py

```diff
@@ -321,9 +310,21 @@ def handle_new_tasks_after_processing(self, task: Task) -> None:
                     )
                 child_task.parent = task
                 child_task.depth = task.depth + 1
+
+                # On retry, we need to see if these child tasks exist.
+                if child_task in task.children:
+                    continue
+                for child in task.children:
+                    if self.is_similar_to_task(child, child_task):
+                        continue
+
                 task.children.append(child_task)
                 if not self.should_skip_task(child_task):
                     self.priority_queue.push(child_task)
+                if max_depth is not None and child_task.depth <= max_depth:
+                    chatter.get().chat_simple(
```

Comment:
Add the ability for this to capture child tasks added and print in a nice markdown thing that is expandable once we get that in the UI